backend/WebApplication1/Infrastructure/Debugging/FileLogger.cs

```diff
@@ -8,38 +8,52 @@ namespace WebApplication1.Infrastructure.Debugging
 {
     public class FileLogger : ILogger, IDisposable
     {
-        private static readonly string _mainLogFileName = AppSettings.Configuration.MainLogFileName;
-        private Lazy<FileLogger> _mainLogger = new Lazy<FileLogger>(() => new FileLogger(_mainLogFileName));
-        public FileLogger MainLogger
+        private readonly string _fileName;
+        private StreamWriter Writer { get; }
+        private Lazy<FailLogger> _mainLogger = new Lazy<FailLogger>(() => new FailLogger());
+        private FailLogger MainLogger
         {
             get
             {
                 return _mainLogger.Value;
             }
         }
 
-        private string _fileName;
-        private Lazy<StreamWriter> _streamWriter;
-        private StreamWriter Writer
+        private class FailLogger : ILogger
         {
-            get
+            // If we for some reason can't log, let's try to log the fact that we can't log!
+            private static readonly string _mainLogFileName = AppSettings.Configuration.MainLogFileName;
+            StreamWriter _writer;
+            public void Log(LogType type, string message)
             {
+                string messageType = Enum.GetName(typeof(LogType), type);
+                string time = DateTime.Now.GetFormattedDateAndTime();
                 try
                 {
-                    return _streamWriter.Value;
+                    _writer = new StreamWriter(_mainLogFileName);
+                    _writer.WriteLine($"[{messageType}][{time}]: {message}");
+                }
+                catch(Exception e) {
+                    var a = e;
```

Comment:
Not entirely convinced that this is a meaningful way of dealing with exception but I'll let it slide.

Reply:
"Not entirely" xd

Reply:
We can't try to log everything until there is some disk space left, we have to end the cycle somehow, so this is meaningful IMO, even tho we don't programmatically write some code